test-suite/Makefile

```diff
@@ -242,6 +250,51 @@ $(addsuffix .log,$(wildcard bugs/closed/*.v)): %.v.log: %.v
 	  fi; \
 	} > "$@"
 
+#######################################################################
+# Unit tests
+#######################################################################
+
+OCAMLOPT := $(OCAMLFIND) opt $(CAMLFLAGS)
+SYSMOD:=-package num,str,unix,dynlink,threads
+
+UNIT_COQDIRS=-I $(LIB)/clib -I $(LIB)/lib -I $(LIB)/kernel -I $(LIB)/kernel/byterun \
+-I $(LIB)/library -I $(LIB)/engine -I $(LIB)/pretyping -I $(LIB)/interp \
+-I $(LIB)/proofs -I $(LIB)/parsing -I $(LIB)/printing -I $(LIB)/tactics -I $(LIB)/vernac \
+-I $(LIB)/stm -I $(LIB)/toplevel
+
+# ML files from unit-test framework, not containing tests
+UNIT_SRCFILES=$(shell find ./unit-tests/src -name *.ml)
+UNIT_ALLMLFILES:=$(shell find ./unit-tests -name *.ml)
+UNIT_MLFILES:=$(filter-out $(UNIT_SRCFILES),$(UNIT_ALLMLFILES))
+UNIT_LOGFILES:=$(patsubst %.ml,%.ml.log,$(UNIT_MLFILES))
+
+UNIT_CMXS=utest.cmx
+
+# this is same link order as CORECMA in Makefile.common in Coq root dir
+UNIT_COQCMXAS=clib.cmxa lib.cmxa kernel.cmxa library.cmxa engine.cmxa pretyping.cmxa \
+ interp.cmxa proofs.cmxa parsing.cmxa printing.cmxa tactics.cmxa vernac.cmxa stm.cmxa toplevel.cmxa
+
+unit-tests/src/utest.cmx : unit-tests/src/utest.ml unit-tests/src/utest.cmi
+	$(SHOW) 'OCAMLOPT  $<'
+	$(HIDE)$(OCAMLOPT) -c -I unit-tests/src -package oUnit $<
+unit-tests/src/utest.cmi : unit-tests/src/utest.mli
+	$(SHOW) 'OCAMLOPT  $<'
+	$(HIDE)$(OCAMLOPT) -package oUnit $<
+
+$(UNIT_LOGFILES) : unit-tests/src/utest.cmx
+
+unit-tests: $(UNIT_LOGFILES)
+
+# Build executable, run it to generate log file
+unit-tests/%.ml.log: unit-tests/%.ml
+	$(SHOW) 'TEST      $<'
+	$(HIDE)$(OCAMLOPT) -linkall -linkpkg -I vernac -I toplevel \
+	                   -I kernel/byterun/ -cclib -lcoqrun \
```

Comment:
This has got to be wrong, vernac, toplevel and kernel/byterun are not present in the test-suite.
Could you try with those `-I`s removed?
Maybe @gares or @ejgallego have a clearer idea of what's going on than me.

Reply:
They are not needed.